File: StructuresAndAlgogrithms/lab1/1.c
{$mode objfpc}
const MAX=10;
	  MIN=1;	
var a:array[1..10]of string=
(
'0001001000',
'0010010000',
'0001010111',
'0000001100',
'0001100000',
'0000101111',
'0000101000',
'0000101000',
'0001101100',
'0010000100'
);

procedure fill(x,y:integer);
begin
	if (x<MIN)or(x>MAX)or(y<MIN)or(y>MAX)or(a[y][x]<>'0')
		then exit;
	a[y][x]:='*';
	fill(x+1,y);
	fill(x-1,y);
	fill(x,y+1);
	fill(x,y-1);	
end;

procedure outa();
	var s:string;
begin
	for s in a do writeln(s);
end;

BEGIN
 fill(5,1);
 outa();
END.

File: StructuresAndAlgorithms/tasks/Examples_lab1/1.pas
{$mode objfpc}
//Если n=1, то n!=1
//иначе n!=n*(n-1)!
function f(n:integer):integer;
begin
	//writeln(ptruint(@n));
	if n=1 then exit(1)
	else exit(n*f(n-1));
end;
var x:int64=-2147483648*34;
BEGIN
 writeln(f(33));	
{ while x<>0 do begin
	write(abs(x mod 2));
	x:=x div 2;
 end;
}	
END.

